src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndex.java

```diff
@@ -30,38 +27,272 @@
  * if that's what you want to do.
  */
 public final class BwaMemIndex implements AutoCloseable {
+
+    private static final int MAXIMUM_NUMBER_OF_CHARACTER_BEFORE_FIRST_HEADER_IN_FASTA_FILES =  4092;
+    private static final char FASTA_HEADER_PREFIX_CHAR = '>';
+
+    public static final List<String> INDEX_FILE_EXTENSIONS =
+            Collections.unmodifiableList(Arrays.asList(
+                    ".amb", ".ann", ".bwt", ".pac", ".sa" ));
+
+    public static final String IMAGE_FILE_EXTENSION = ".img";
+
+    public static final List<String> FASTA_FILE_EXTENSIONS =
+            Collections.unmodifiableList(Arrays.asList(".fasta", ".fa"));
+
+    /**
+     * Indexing algorithms supported by Bwa.
+     */
+    public enum Algorithm {
+
+        /**
+         * Chooses the most appropriate algorithm based on characteristic
+         * of the reference. For references shorter than 2Gbases it will use {@link #IS}
+         * whereas for larger reference it will employ {@link #RB2}.
+         */
+        AUTO,
+
+        /**
+          * Linear-time algorithm for constructing suffix array.
+          * It requires 5.37N memory where N is the size of the database.
+          * IS is moderately fast, but does not work with database larger than 2GB.
+          * IS is the default algorithm due to its simplicity. The current codes for
+          * IS algorithm are reimplemented by Yuta Mori.
+          *
+          * @see "http://bio-bwa.sourceforge.net/bwa.shtml"
+          */
+        IS,
+
+        /**
+         * The <i>Ropebwt2</i> algorithm.
+         *
+         * @see "https://arxiv.org/pdf/1406.0426.pdf"
+         */
+        RB2;
+        /**
+         * The string name use by the Bwa command line to denote that algorithm.
+         * @return never {@code null} and unique across algorithms.
+         */
+        public String toBwaName() {
+            return this.toString().toLowerCase();
+        }
+    }
+
     private final String indexImageFile; // stash this for error messages
     private volatile long indexAddress; // address where the index was memory-mapped (for use by C code)
     private final AtomicInteger refCount; // keep track of how many threads are actively aligning
     private final List<String> refContigNames; // the reference dictionary from the index
     private static volatile boolean nativeLibLoaded = false; // whether we've loaded the native library or not
 
+    public static void createIndexImageFromFastaFile(final String fasta) {
```

Comment:
javadoc explaining to which method it delegates, the default image file name and default algorithm